keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -1,113 +1,560 @@
+- If distributions do not intend to support tokens signed by external signers after the feature is disabled, then they can come up with a fair warning.
+
+###### What happens if we re-enable the feature if it was previously rolled back?
+
+experience would be the same as when enabling.
+
+###### Are there any tests for feature enablement/disablement?
+
+- Unit and E2E tests will be added.
+- The tests would **not** focus on continuing to support the same key set when enabling/disabling but would rather focus on viability of enabling/disabling.
+
+### Rollout, Upgrade and Rollback Planning
+
+Since it is an opt-in feature; upgrading existing clusters to versions with this feature shall be a no-op in context of the feature.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Since it is an opt-in feature, existing clusters shall remain unaffected. However, since we are touching paths for JWT signing and authentication, if a bug slips in, it can disrupt the respective operation.
 
-New token generation and validation could suffer a performance difference when reading over a socket, as an external process will be signing data.
+###### What specific metrics should inform a rollback?
 
-Signing and verifying tokens over a grpc API carries the risk of a server side request forgery, where a malicious client could generate tokens. To mitigate this risk, the API will only be accessible over a unix socket.
+1. High error rates on service account token creation.
 
-## Graduation Criteria
+   - Check for `apiserver_request_total`
+   - Use following labels: resource="serviceaccounts", subresource="token"
+   - Compute error rate:
 
-<!-- TODO -->
+     ````
+      sum(rate(apiserver_request_total{job="kubernetes-apiservers",group="", version="v1",resource="serviceaccounts",subresource="token",code=~"5.."}[5m]))
+     ````
+
+2. High usage of stale tokens being reported
+
+   - Check for `serviceaccount_stale_tokens_total`
+   - Compare against `serviceaccount_valid_tokens_total`
+
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Will be covered by integration tests.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The Feature would not be used by workload directly but will be used by kube-apiserver.
+
+The usage should be visible to the operator using Audit logs.
+<!-- TODO 
+  Add details on increasing audit log surface area for External signers
+-->
+
+###### How can someone using this feature know that it is working for their instance?
+
+The feature is not used by individual pods but the kube-apiserver itself. Simply successful Token requests and successful auth with kube-apiserver when using a bearer token in the request shall be indicators enough. 
```

Comment:
Will we have a `/healthz/` endpoint for this feature?

Reply:
was discussed as part of https://github.com/kubernetes/enhancements/pull/4852#discussion_r1783239827

we decided to rely on status as reported by kube-apiserver since fetching JWKs falls in critical path.

Reply:
the initial read of public keys should inform healthz / readyz, so there will be an endpoint for that initial sync ... once the first read happens successfully, subsequent failures wouldn't turn the API server unhealth, but they would be observable via metrics

Reply:
Also mentioned it in this section.